src/tests/test_numerictype_cast.cpp

```diff
@@ -0,0 +1,118 @@
+#include <memory>
+
+#include <gtest/gtest.h>
+
+#include "types.h"
+
+namespace tests {
+struct NumericCast : public ::testing::Test {
+protected:
+    void SetUp() override {
+        dbl_b_ptr.reset(new model::Double(2.79));
+        dbl_a_ptr.reset(new model::Double(5.58));
+        int_a_ptr.reset(new model::Int(21));
+        int_b_ptr.reset(new model::Int(7));
+    }
+    void TearDown() override {
+        dbl_b_ptr.reset(new model::Double(2.79));
+        dbl_a_ptr.reset(new model::Double(5.58));
+        int_a_ptr.reset(new model::Int(21));
+        int_b_ptr.reset(new model::Int(7));
+    }
+    std::unique_ptr<model::Double> dbl_a_ptr;
+    std::unique_ptr<model::Double> dbl_b_ptr;
+    std::unique_ptr<model::Int> int_a_ptr;
+    std::unique_ptr<model::Int> int_b_ptr;
+
+    model::DoubleType dbl_type;
+    model::INumericType& dbl_type_ref = dbl_type;
```

Comment:
`dbl_type` is not used anywhere in these tests, only to create a variable to reference to. This could be achieved using `const` reference without creating additional variables:
```cpp
model::INumericType const& dbl_type = model::DoubleType{};
```